compiler-plugin/src/main/java/io/ballerina/stdlib/email/compiler/EmailServiceValidator.java

```diff
@@ -276,6 +279,24 @@ private void validateParameter(SeparatedNodeList<ParameterNode> parameterNodes,
         }
     }
 
+    private boolean validateOnMessageParams(FunctionDefinitionNode functionDefinitionNode) {
+        FunctionTypeSymbol functionTypeSymbol = ((MethodSymbol) ctx.semanticModel().symbol(functionDefinitionNode)
+                .get()).typeDescriptor();
+        List<ParameterSymbol> inputParams = functionTypeSymbol.params().get();
+        String moduleId = getModuleId(inputParams.get(0));
+        String typeDescriptorSignature = inputParams.get(0).typeDescriptor().signature();
```

Comment:
Here we have only validate the first parameter. Don't we have to validate whether the developer has provided more than one parameter ?

Reply:
Validation added.